tests/utils/test_runner/conftest.py

```diff
@@ -0,0 +1,462 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""
+Basic high-level plugin file for pytest.
+
+See [Writing plugins](https://docs.pytest.org/en/latest/writing_plugins.html)
+for more information.
+
+This plugin adds the following command-line options:
+
+* `--test_conf` - Path to test configuration file. Used to parametrize tests.
+  Format: YAML file.
+* `--exe` - Path to a test binary to execute.
+* `--niter` - Number of times to run executable.
+"""
+
+import hashlib
+import json
+import logging
+# pylint:disable=import-error
+import os
+import shutil
+import sys
+import tempfile
+import pytest
+import yaml
+from copy import deepcopy
+
+from pathlib import Path
+from jsonschema import validate, ValidationError
+
+UTILS_DIR = Path(__file__).parent.parent
+sys.path.insert(0, str(UTILS_DIR))
+
+from scripts.run_test import check_positive_int
+from test_runner.utils import upload_data, metadata_from_manifest, query_timeline, DATABASES, DB_COLLECTIONS
+from platform_utils import get_os_name, get_os_version, get_cpu_info
+
+
+# -------------------- CLI options --------------------
+
+
+def pytest_addoption(parser):
+    """Specify command-line options for all plugins"""
+    test_args_parser = parser.getgroup("test run")
+    test_args_parser.addoption(
+        "--test_conf",
+        type=Path,
+        help="path to a test config",
+        default=Path(__file__).parent / "test_config.yml"
+    )
+    test_args_parser.addoption(
+        "--exe",
+        required=True,
+        dest="executable",
+        type=Path,
+        help="path to a test binary to execute"
+    )
+    test_args_parser.addoption(
+        "--niter",
+        type=check_positive_int,
+        help="number of iterations to run executable and aggregate results",
+        default=3
+    )
+    helpers_args_parser = parser.getgroup("test helpers")
+    helpers_args_parser.addoption(
+        "--dump_refs",
+        type=Path,
+        help="path to dump test config with references updated with statistics collected while run",
+    )
+    helpers_args_parser.addoption(
+        '--timeline_report',
```

Comment:
requires connection to db, so should be moved to `db_args_parser` with `required=is_db_used` check